// pro-gallery-lib/dist/cjs/core/helpers/layoutHelper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.removeBordersIfNeeded = exports.processNumberOfImagesPerCol = exports.processGridStyle = exports.processNumberOfImagesPerRow = exports.calcTargetItemSize = void 0;
var tslib_1 = require("tslib");
/* eslint-disable prettier/prettier */
var utils_1 = tslib_1.__importDefault(require("../../common/utils"));
var windowWrapper_1 = tslib_1.__importDefault(require("../../common/window/windowWrapper"));
var placements_1 = tslib_1.__importStar(require("../../common/constants/placements"));
var infoBehaviourOnHover_1 = tslib_1.__importDefault(require("../../common/constants/infoBehaviourOnHover"));
var loadingMode_1 = tslib_1.__importDefault(require("../../common/constants/loadingMode"));
var loadingWithColorMode_1 = tslib_1.__importDefault(require("../../common/constants/loadingWithColorMode"));
var slideAnimations_1 = tslib_1.__importDefault(require("../../common/constants/slideAnimations"));
var gallerySizeType_1 = tslib_1.__importDefault(require("../../common/constants/gallerySizeType"));
var arrowsPosition_1 = tslib_1.__importDefault(require("../../common/constants/arrowsPosition"));
var index_1 = tslib_1.__importDefault(require("../../common/constants/index"));
var optionsUtils_1 = require("./optionsUtils");
var textBoxDimensionsHelper_1 = tslib_1.__importDefault(require("./textBoxDimensionsHelper"));
var slideAnimation_1 = tslib_1.__importDefault(require("../../settings/options/slideAnimation"));
var arrowsPosition_2 = tslib_1.__importDefault(require("../../settings/options/arrowsPosition"));
var calcTargetItemSize = function (options, smartCalc) {
    if (smartCalc === void 0) { smartCalc = false; }
    if (options.gallerySizeType === gallerySizeType_1.default.PIXELS &&
        options.gallerySizePx > 0) {
        return options.gallerySizePx;
    }
    else if (options.gallerySizeType === gallerySizeType_1.default.RATIO &&
        options.gallerySizeRatio > 0) {
        return (((windowWrapper_1.default && windowWrapper_1.default.innerWidth) || 980) * (options.gallerySizeRatio / 100));
    }
    else {
        return smartCalc ? smartCalc : options.gallerySize;
    }
};
exports.calcTargetItemSize = calcTargetItemSize;
var processNumberOfImagesPerRow = function (options) {
    //This will be used in the masonry and grid presets
    var res = tslib_1.__assign({}, options);
    res.fixedColumns = 0;
    if (options.scrollDirection === index_1.default.scrollDirection.VERTICAL || //relevant for grid, in Masonry its fixed to !oneRow
        options.isVertical //relevant for masonry, in grid its fixed to true.
    ) {
        res.fixedColumns =
            String(options.gridStyle) === '1'
                ? Number(options.numberOfImagesPerRow)
                : 0;
        res.groupTypes = '1';
        res.groupSize = 1;
        res.collageAmount = 0;
        res.collageDensity = 0;
    }
    return res;
};
exports.processNumberOfImagesPerRow = processNumberOfImagesPerRow;
var processGridStyle = function (options) {
    var res = tslib_1.__assign({}, options);
    res.gridStyle =
        res.isVertical ? options.gridStyle : index_1.default.gridStyle.FIT_TO_SCREEN;
    return res;
};
exports.processGridStyle = processGridStyle;
var processNumberOfImagesPerCol = function (options) {
    //This will be used in the grid preset
    var res = tslib_1.__assign({}, options);
    if (!utils_1.default.isUndefined(options.numberOfImagesPerCol) &&
        options.scrollDirection === index_1.default.scrollDirection.HORIZONTAL) {
        res.fixedColumns = 0;
        switch (options.numberOfImagesPerCol) {
            case 1:
            default:
                res.groupTypes = '1';
                res.groupSize = 1;
                res.collageAmount = 0;
                res.collageDensity = 0;
                break;
            case 2:
                res.groupTypes = '2v';
                res.groupSize = 2;
                res.collageAmount = 1;
                res.collageDensity = 1;
                break;
            case 3:
                res.groupTypes = '3v';
                res.groupSize = 3;
                res.collageAmount = 1;
                res.collageDensity = 1;
                break;
        }
    }
    return res;
};
exports.processNumberOfImagesPerCol = processNumberOfImagesPerCol;
var forceInfoOnHoverWhenNeeded = function (options) {
    var _options = tslib_1.__assign({}, options);
    if (!index_1.default.isLayout('SLIDER')(_options) && //not slider
        !index_1.default.isLayout('COLUMN')(_options) && //not columns
        !index_1.default.isLayout('SLIDESHOW')(_options) //not columns
    ) {
        if ((!_options.isVertical || //layout orientation is horizontal
            _options.groupSize > 1) //groups are larger than one (items can be on top or right left of eachother)
        ) {
            // Dont allow titlePlacement to be above / below / left / right
            _options.titlePlacement = placements_1.default.SHOW_ON_HOVER;
        }
    }
    return _options;
};
var setMobileFonts = function (options) {
    var _options = tslib_1.__assign({}, options);
    if (isSlideshowFont(_options)) {
        if (!utils_1.default.isUndefined(_options.itemFontSlideshow)) {
            _options = setTextUnderline('itemFontSlideshow', 'textDecorationTitle', _options);
        }
        if (!utils_1.default.isUndefined(_options.itemDescriptionFontSlideshow)) {
            _options = setTextUnderline('itemDescriptionFontSlideshow', 'textDecorationDesc', _options);
        }
    }
    else {
        if (!utils_1.default.isUndefined(_options.itemFont)) {
            _options = setTextUnderline('itemFont', 'textDecorationTitle', _options);
        }
        if (!utils_1.default.isUndefined(_options.itemDescriptionFont)) {
            _options = setTextUnderline('itemDescriptionFont', 'textDecorationDesc', _options);
        }
    }
    return _options;
};
var forceHoverToShowTextsIfNeeded = function (options) {
    var _options = tslib_1.__assign({}, options);
    if (options.EXPERIMENTALallowParallelInfos)
        return _options;
    if (!placements_1.hasHoverPlacement(_options.titlePlacement) &&
        _options.hoveringBehaviour !== infoBehaviourOnHover_1.default.NEVER_SHOW) {
        _options.hoveringBehaviour = infoBehaviourOnHover_1.default.APPEARS;
    }
    return _options;
};
var blockScrollOnFadeOrDeckScrollAnimations = function (options) {
    var _options = tslib_1.__assign({}, options);
    if ((options.slideAnimation === index_1.default.slideAnimations.FADE ||
        options.slideAnimation === index_1.default.slideAnimations.DECK) && (slideAnimation_1.default.isRelevant(options))) {
        _options.enableScroll = false;
    }
    return _options;
};
var blockVideoControlsOnMouseCursorNavigation = function (options) {
    var _options = tslib_1.__assign({}, options);
    if ((options.arrowsPosition === index_1.default.arrowsPosition.MOUSE_CURSOR) && (arrowsPosition_2.default.isRelevant(options))) {
        _options.showVideoControls = false;
    }
    return _options;
};
var processImageLoadingWithColorMode = function (options) {
    var _options = tslib_1.__assign({}, options);
    if (_options.imageLoadingMode === loadingMode_1.default.COLOR &&
        _options.imageLoadingWithColorMode ===
            loadingWithColorMode_1.default.MAIN_COLOR) {
        _options.imageLoadingMode = loadingMode_1.default.MAIN_COLOR;
    }
    return _options;
};
var removeShadowOnHorizontalGalleries = function (options) {
    var _options = tslib_1.__assign({}, options);
    if (_options.itemEnableShadow && _options.scrollDirection === index_1.default.scrollDirection.HORIZONTAL) {
        _options.itemEnableShadow = false;
    }
    return _options;
};
var forceHorizontalOrientationInHorizontalGalleries = function (options) {
    var _options = tslib_1.__assign({}, options);
    if (_options.scrollDirection === index_1.default.scrollDirection.HORIZONTAL) {
        // in horizontal galleries allow only horizontal orientation
        _options.isVertical = false;
    }
    return _options;
};
var removeLoopOnVerticalGalleries = function (options) {
    var _options = tslib_1.__assign({}, options);
    if (_options.scrollDirection === index_1.default.scrollDirection.VERTICAL) {
        _options.slideshowLoop = false; // allow slideshowLoop only for horizontal layouts
    }
    return _options;
};
var limitImageMargin = function (options) {
    var _options = tslib_1.__assign({}, options);
    if (_options.imageMargin > 0) {
        _options.imageMargin = Math.min(_options.imageMargin, 50); // limit mobile spacing to 50px (25 on each side)
    }
    return _options;
};
var forceScrollAnimationOnSingleImageInViewGalleries = function (options) {
    var _options = tslib_1.__assign({}, options);
    if (_options.scrollDirection === index_1.default.scrollDirection.VERTICAL ||
        _options.groupSize > 1 ||
        !_options.cubeImages) {
        _options.slideAnimation = slideAnimations_1.default.SCROLL;
    }
    return _options;
};
var removeArrowPaddingIfOutsideTheGallery = function (options) {
    var _options = tslib_1.__assign({}, options);
    if (_options.arrowsPosition === arrowsPosition_1.default.OUTSIDE_GALLERY) {
        _options.arrowsPadding = 0;
    }
    return _options;
};
var processForceMobileCustomButton = function (options) {
    var _options = tslib_1.__assign({}, options);
    if (_options.forceMobileCustomButton) {
        _options.targetItemSize = Math.round(30 * 8.5 + 150);
        _options.titlePlacement = placements_1.default.SHOW_BELOW;
        _options.galleryLayout = 2;
        _options.fixedColumns = 1;
        _options.numberOfImagesPerRow = 1;
    }
    return _options;
};
var processSpecialGallerySize = function (options) {
    var _options = tslib_1.__assign({}, options);
    // in case a special gallery size was specified, use it
    if (_options.gallerySizeType === gallerySizeType_1.default.PIXELS &&
        _options.gallerySizePx > 0) {
        _options.targetItemSize = _options.gallerySizePx;
    }
    else if (_options.gallerySizeType === gallerySizeType_1.default.RATIO &&
        _options.gallerySizeRatio > 0) {
        _options.targetItemSize =
            ((windowWrapper_1.default && windowWrapper_1.default.innerWidth) || 980) *
                (_options.gallerySizeRatio / 100);
    }
    return _options;
};
var processLoadMoreButtonFont = function (options) {
    var _options = tslib_1.__assign({}, options);
    if (_options.loadMoreButtonFont && utils_1.default.isMobile()) {
        _options.loadMoreButtonFont.value =
            _options.loadMoreButtonFont.value.replace(/^font\s*:\s*/, '');
        _options.loadMoreButtonFont.value =
            _options.loadMoreButtonFont.value.replace(/;$/, '');
        if (_options.loadMoreButtonFont.value.indexOf('underline') > -1) {
            _options.loadMoreButtonFont.value =
                _options.loadMoreButtonFont.value.replace('underline', '');
            _options.textDecorationLoadMore = 'underline';
        }
        else {
            _options.textDecorationLoadMore = 'none';
        }
    }
    return _options;
};
var blockCounterByProduct = function (options) {
    var _options = tslib_1.__assign({}, options);
    if (!_options.allowSlideshowCounter) {
        return _options;
    }
    if (!_options.isAutoSlideshow) {
        _options.allowSlideshowCounter = false;
    }
    if (index_1.default.isLayout('SLIDESHOW')(options) === false) {
        _options.allowSlideshowCounter = false;
    }
    return _options;
};
var addMarginsToSupportShadows = function (options) {
    var _options = tslib_1.__assign({}, options);
    if (_options.itemEnableShadow && _options.scrollDirection === index_1.default.scrollDirection.VERTICAL) {
        // add gallerySpacing to allow the shadow to be seen
        var _gallerySpacing = Math.max(_options.layoutParams.gallerySpacing, (_options.itemShadowSize || 0) +
            (_options.itemShadowBlur || 0));
        _options = optionsUtils_1.assignByString(_options, 'layoutParams_gallerySpacing', _gallerySpacing);
    }
    return _options;
};
var centerArrowsWhenNeeded = function (options) {
    var _options = tslib_1.__assign({}, options);
    var isSingleVerticalItemRendered = _options.layoutParams.repeatingGroupTypes ?
        _options.layoutParams.repeatingGroupTypes == "1" : _options.groupTypes == "1"; // only one item is being rendered vertically?
    var filteredPlacement = _options.titlePlacement // filtering hover since it doesn't affect this product
        .split(',')
        .filter(function (placement) { return !placements_1.isHoverPlacement(placement); })
        .join(',');
    if (!placements_1.isConstantVerticalPlacement(filteredPlacement) ||
        !isSingleVerticalItemRendered) {
        // if text (info) placement is not above/below placement or more then 1 images per col, arrows are gallery("item") centered
        _options.arrowsVerticalPosition = index_1.default.arrowsVerticalPosition.ITEM_CENTER;
    }
    return _options;
};
var removeBordersIfNeeded = function (options) {
    //TODO this can go into the _optionspective 4 layouts.
    var _options = tslib_1.__assign({}, options);
    if (_options.cubeType === 'fit') {
        _options.itemBorderWidth = 0;
        _options.itemBorderRadius = 0;
        _options.itemEnableShadow = false;
    }
    return _options;
};
exports.removeBordersIfNeeded = removeBordersIfNeeded;
var setTextUnderline = function (itemFontOption, textDecorationType, options) {
    /* itemFontOption: itemFontSlideshow / itemDescriptionFontSlideshow / itemFont / itemDescriptionFont
    textDecorationType: textDecorationTitle / textDecorationDesc */
    var _options = tslib_1.__assign({}, options);
    _options[itemFontOption].value = _options[itemFontOption].value.replace(/^font\s*:\s*/, '');
    _options[itemFontOption].value = _options[itemFontOption].value.replace(/;$/, '');
    if (_options[itemFontOption].value.indexOf('underline') > -1 ||
        _options[itemFontOption].style.underline === true) {
        _options[itemFontOption].value = _options[itemFontOption].value.replace('underline', '');
        _options[textDecorationType] = 'underline';
    }
    else if (_options[itemFontOption].style.underline === false) {
        _options[textDecorationType] = 'none';
    }
    return _options;
};
function processLayouts(options, customExternalInfoRendererExists) {
    var processedOptions = tslib_1.__assign({}, options);
    if (utils_1.default.isMobile()) {
        processedOptions = setMobileFonts(processedOptions);
        processedOptions = limitImageMargin(processedOptions);
    }
    processedOptions = forceInfoOnHoverWhenNeeded(processedOptions);
    processedOptions = forceHoverToShowTextsIfNeeded(processedOptions);
    processedOptions = processImageLoadingWithColorMode(processedOptions);
    processedOptions = removeShadowOnHorizontalGalleries(processedOptions);
    processedOptions = addMarginsToSupportShadows(processedOptions);
    processedOptions = removeArrowPaddingIfOutsideTheGallery(processedOptions);
    processedOptions = forceHorizontalOrientationInHorizontalGalleries(processedOptions);
    processedOptions = removeLoopOnVerticalGalleries(processedOptions);
    processedOptions = forceScrollAnimationOnSingleImageInViewGalleries(processedOptions);
    processedOptions = processLoadMoreButtonFont(processedOptions); //contains if isMobile, but also has an else.
    processedOptions = processForceMobileCustomButton(processedOptions); //TODO this seems like it doesnt really exists. consider deleting support.
    processedOptions = processSpecialGallerySize(processedOptions);
    processedOptions = textBoxDimensionsHelper_1.default(processedOptions, customExternalInfoRendererExists);
    processedOptions = centerArrowsWhenNeeded(processedOptions);
    processedOptions = blockCounterByProduct(processedOptions);
    processedOptions = blockScrollOnFadeOrDeckScrollAnimations(processedOptions);
    processedOptions = blockVideoControlsOnMouseCursorNavigation(processedOptions);
    return processedOptions;
}
function isSlideshowFont(options) {
    if (placements_1.hasExternalVerticalPlacement(options.titlePlacement)) {
        if (index_1.default.isLayout('SLIDER')(options) || index_1.default.isLayout('PANORAMA')(options) || index_1.default.isLayout('COLUMN')(options)) {
            return true;
        }
        else if (index_1.default.isLayout('MASONRY')(options) && options.isVertical) {
            return true;
        }
        else if (index_1.default.isLayout('GRID')(options) && options.scrollDirection !== 1) {
            return true;
        }
    }
    return false;
}
exports.default = processLayouts;
/* eslint-enable prettier/prettier */
