// pro-gallery-lib/dist/cjs/core/helpers/migratorStore.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.reverseBooleanTo = exports.namingChange = exports.reverseBooleans = exports.changeNames = exports.nameChangedStylingParams = exports.reversedBehaviourParams = exports.nameChangedBehaviourParams = exports.reversedLayoutParams = exports.nameChangedLayoutParams = void 0;
var tslib_1 = require("tslib");
var optionsMap_1 = tslib_1.__importDefault(require("./optionsMap"));
exports.nameChangedLayoutParams = [
    ['galleryLayout', optionsMap_1.default.layoutParams.structure.galleryLayout],
    ['imageMargin', optionsMap_1.default.layoutParams.structure.itemSpacing],
    ['groupsPerStrip', optionsMap_1.default.layoutParams.groups.numberOfGroupsPerRow],
    ['cubeImages', optionsMap_1.default.layoutParams.crop.enable],
    ['smartCrop', optionsMap_1.default.layoutParams.crop.enableSmartCrop],
    ['cropOnlyFill', optionsMap_1.default.layoutParams.crop.cropOnlyFill],
    ['minItemSize', optionsMap_1.default.layoutParams.targetItemSize.minimum],
    ['scatter', optionsMap_1.default.layoutParams.structure.scatter.randomScatter],
    ['rotatingScatter', optionsMap_1.default.layoutParams.structure.scatter.manualScatter],
    ['numberOfImagesPerCol', optionsMap_1.default.layoutParams.structure.numberOfGridRows],
    //['collageAmount', optionsMap.layoutParams.groups.amount], //This doesnt really exist. need to eradicate as a refactor
    ['collageDensity', optionsMap_1.default.layoutParams.groups.density],
    ['chooseBestGroup', optionsMap_1.default.layoutParams.groups.groupByOrientation],
    ['groupSize', optionsMap_1.default.layoutParams.groups.groupSize],
    ['hasThumbnails', optionsMap_1.default.layoutParams.thumbnails.enable],
    ['thumbnailSpacings', optionsMap_1.default.layoutParams.thumbnails.spacing],
    ['thumbnailSize', optionsMap_1.default.layoutParams.thumbnails.size],
    ['showArrows', optionsMap_1.default.layoutParams.navigationArrows.enable],
    ['arrowsPadding', optionsMap_1.default.layoutParams.navigationArrows.padding],
    [
        'arrowsVerticalPosition',
        optionsMap_1.default.layoutParams.navigationArrows.verticalAlignment,
    ],
    ['arrowsSize', optionsMap_1.default.layoutParams.navigationArrows.size],
    ['imageInfoType', optionsMap_1.default.layoutParams.info.layout],
    //['textBoxWidthPercent', optionsMap.layoutParams.info.widthByPercent],
    ['textImageSpace', optionsMap_1.default.layoutParams.info.spacing],
    ['textBoxBorderWidth', optionsMap_1.default.layoutParams.info.border.width],
    ['textBoxBorderColor', optionsMap_1.default.layoutParams.info.border.color],
    ['textBoxBorderRadius', optionsMap_1.default.layoutParams.info.border.radius],
];
exports.reversedLayoutParams = [
    ['useMaxDimensions', optionsMap_1.default.layoutParams.structure.enableStreching],
];
exports.nameChangedBehaviourParams = [
    [
        'magnificationLevel',
        optionsMap_1.default.behaviourParams.item.content.magnificationValue,
    ],
    ['videoLoop', optionsMap_1.default.behaviourParams.item.video.loop],
    [
        'showVideoPlayButton',
        optionsMap_1.default.behaviourParams.item.video.enablePlayButton,
    ],
    ['showVideoControls', optionsMap_1.default.behaviourParams.item.video.enableControls],
    [
        'enableVideoPlaceholder',
        optionsMap_1.default.behaviourParams.item.video.enablePlaceholder,
    ],
    ['overlayAnimation', optionsMap_1.default.behaviourParams.item.overlay.hoverAnimation],
    ['overlayPosition', optionsMap_1.default.behaviourParams.item.overlay.position],
    ['overlaySize', optionsMap_1.default.behaviourParams.item.overlay.size],
    ['overlaySizeType', optionsMap_1.default.behaviourParams.item.overlay.sizeUnits],
    ['overlayPadding', optionsMap_1.default.behaviourParams.item.overlay.padding],
    [
        'imageHoverAnimation',
        optionsMap_1.default.behaviourParams.item.content.hoverAnimation,
    ],
    [
        'imagePlacementAnimation',
        optionsMap_1.default.behaviourParams.item.content.placementAnimation,
    ],
    ['imageLoadingMode', optionsMap_1.default.behaviourParams.item.content.loader],
    [
        'scrollSnap',
        optionsMap_1.default.behaviourParams.gallery.horizontal.enableScrollSnap,
    ],
    ['scrollAnimation', optionsMap_1.default.behaviourParams.gallery.scrollAnimation],
    [
        'shouldIndexDirectShareLinkInSEO',
        optionsMap_1.default.behaviourParams.gallery.enableIndexingShareLinks,
    ],
    [
        'loadMoreButtonText',
        optionsMap_1.default.behaviourParams.gallery.vertical.loadMore.text,
    ],
    [
        'slideAnimation',
        optionsMap_1.default.behaviourParams.gallery.horizontal.slideAnimation,
    ],
    [
        'slideTransition',
        optionsMap_1.default.behaviourParams.gallery.horizontal.slideTransition,
    ],
    [
        'scrollDuration',
        optionsMap_1.default.behaviourParams.gallery.horizontal.navigationDuration,
    ],
    ['slideshowLoop', optionsMap_1.default.behaviourParams.gallery.horizontal.loop],
    [
        'autoSlideshowInterval',
        optionsMap_1.default.behaviourParams.gallery.horizontal.autoSlide.interval,
    ],
    [
        'pauseAutoSlideshowOnHover',
        optionsMap_1.default.behaviourParams.gallery.horizontal.autoSlide.pauseOnHover,
    ],
    [
        'autoSlideshowContinuousSpeed',
        optionsMap_1.default.behaviourParams.gallery.horizontal.autoSlide.speed,
    ],
    [
        'allowSlideshowCounter',
        optionsMap_1.default.behaviourParams.gallery.horizontal.slideshowInfo.enableCounter,
    ],
    [
        'playButtonForAutoSlideShow',
        optionsMap_1.default.behaviourParams.gallery.horizontal.slideshowInfo
            .enablePlayButton,
    ],
];
exports.reversedBehaviourParams = [
    [
        'enableInfiniteScroll',
        optionsMap_1.default.behaviourParams.gallery.vertical.loadMore.enable,
    ],
    ['allowContextMenu', optionsMap_1.default.behaviourParams.gallery.blockContextMenu],
    ['enableScroll', optionsMap_1.default.behaviourParams.gallery.horizontal.blockScroll],
];
exports.nameChangedStylingParams = [
    ['itemShadowBlur', optionsMap_1.default.stylingParams.itemShadowBlur],
    ['itemShadowDirection', optionsMap_1.default.stylingParams.itemShadowDirection],
    [
        'itemShadowOpacityAndColor',
        optionsMap_1.default.stylingParams.itemShadowOpacityAndColor,
    ],
    ['arrowsColor', optionsMap_1.default.stylingParams.arrowsColor],
    ['itemShadowSize', optionsMap_1.default.stylingParams.itemShadowSize],
    ['itemEnableShadow', optionsMap_1.default.stylingParams.itemEnableShadow],
    ['itemBorderRadius', optionsMap_1.default.stylingParams.itemBorderRadius],
    ['itemBorderWidth', optionsMap_1.default.stylingParams.itemBorderWidth],
];
//---------tooling---------//
function changeNames(obj, pairsArray) {
    var _obj = tslib_1.__assign({}, obj);
    for (var _i = 0, pairsArray_1 = pairsArray; _i < pairsArray_1.length; _i++) {
        var _a = pairsArray_1[_i], oldName = _a[0], newName = _a[1];
        _obj = namingChange(_obj, oldName, newName);
    }
    return _obj;
}
exports.changeNames = changeNames;
function reverseBooleans(obj, pairsArray) {
    var _obj = tslib_1.__assign({}, obj);
    for (var _i = 0, pairsArray_2 = pairsArray; _i < pairsArray_2.length; _i++) {
        var _a = pairsArray_2[_i], oldName = _a[0], newName = _a[1];
        _obj = reverseBooleanTo(_obj, oldName, newName);
    }
    return _obj;
}
exports.reverseBooleans = reverseBooleans;
function namingChange(obj, oldName, newName) {
    if (typeof obj[newName] === 'undefined' &&
        typeof obj[oldName] !== 'undefined') {
        obj[newName] = obj[oldName];
    }
    delete obj[oldName];
    return obj;
}
exports.namingChange = namingChange;
function reverseBooleanTo(obj, oldName, newName) {
    if (typeof obj[newName] === 'undefined' &&
        typeof obj[oldName] !== 'undefined') {
        obj[newName] = !obj[oldName];
    }
    delete obj[oldName];
    return obj;
}
exports.reverseBooleanTo = reverseBooleanTo;
