// pro-gallery/dist/cjs/components/item/itemViewStyleProvider.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getInnerInfoStyle = exports.getOuterInfoStyle = exports.getImageStyle = exports.getContainerStyle = void 0;
var tslib_1 = require("tslib");
var pro_gallery_lib_1 = require("pro-gallery-lib");
function getContainerStyle(options) {
    return tslib_1.__assign({}, ((options.imageInfoType ===
        pro_gallery_lib_1.GALLERY_CONSTS.infoType.ATTACHED_BACKGROUND ||
        pro_gallery_lib_1.GALLERY_CONSTS.hasHoverPlacement(options.titlePlacement)) && tslib_1.__assign(tslib_1.__assign({}, getBorderStyle(options.itemBorderRadius, options.itemBorderWidth, options.itemBorderColor)), boxShadow(options))));
}
exports.getContainerStyle = getContainerStyle;
function boxShadow(options) {
    var _boxShadow = {};
    if (options.itemEnableShadow) {
        var itemShadowBlur = options.itemShadowBlur, itemShadowDirection = options.itemShadowDirection, itemShadowSize = options.itemShadowSize;
        var alpha = ((-1 * (Number(itemShadowDirection) - 90)) / 360) * 2 * Math.PI;
        var shadowX = Math.round(itemShadowSize * Math.cos(alpha));
        var shadowY = Math.round(-1 * itemShadowSize * Math.sin(alpha));
        _boxShadow = {
            boxShadow: shadowX + "px " + shadowY + "px " + itemShadowBlur + "px " + pro_gallery_lib_1.utils.formatColor(options.itemShadowOpacityAndColor),
        };
    }
    return _boxShadow;
}
function getImageStyle(options) {
    return tslib_1.__assign({}, (!pro_gallery_lib_1.GALLERY_CONSTS.hasHoverPlacement(options.titlePlacement) &&
        (options.imageInfoType === pro_gallery_lib_1.GALLERY_CONSTS.infoType.NO_BACKGROUND ||
            options.imageInfoType ===
                pro_gallery_lib_1.GALLERY_CONSTS.infoType.SEPARATED_BACKGROUND) && tslib_1.__assign({}, getBorderStyle(options.itemBorderRadius, options.itemBorderWidth, options.itemBorderColor))));
}
exports.getImageStyle = getImageStyle;
function getBorderStyle(borderRadius, borderWidth, borderColor) {
    return tslib_1.__assign(tslib_1.__assign({ overflow: 'hidden' }, (borderRadius > 0 && { borderRadius: borderRadius })), (borderWidth > 0 && {
        borderWidth: borderWidth + 'px',
        borderColor: pro_gallery_lib_1.utils.formatColor(borderColor),
        borderStyle: 'solid',
    }));
}
function getOuterInfoStyle(placement, options, mediaHeight, textBoxHeight) {
    var styles = tslib_1.__assign(tslib_1.__assign({}, (pro_gallery_lib_1.GALLERY_CONSTS.hasExternalHorizontalPlacement(placement) && {
        height: mediaHeight,
        float: pro_gallery_lib_1.GALLERY_CONSTS.isExternalRightPlacement(placement)
            ? 'right'
            : 'left',
    })), (pro_gallery_lib_1.GALLERY_CONSTS.hasExternalVerticalPlacement(placement) && {
        height: textBoxHeight,
        boxSizing: 'content-box',
    }));
    if (options.imageInfoType === pro_gallery_lib_1.GALLERY_CONSTS.infoType.SEPARATED_BACKGROUND) {
        return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, styles), getBorderStyle(options.textBoxBorderRadius, options.textBoxBorderWidth, options.textBoxBorderColor)), (pro_gallery_lib_1.GALLERY_CONSTS.hasExternalAbovePlacement(placement) && {
            marginBottom: options.textImageSpace,
        })), (pro_gallery_lib_1.GALLERY_CONSTS.hasExternalBelowPlacement(placement) && {
            marginTop: options.textImageSpace,
        }));
    }
    return styles;
}
exports.getOuterInfoStyle = getOuterInfoStyle;
function getInnerInfoStylesAboveOrBelow(options, infoHeight) {
    return {
        width: '100%',
        height: infoHeight,
    };
}
function getInnerInfoStylesRightOrLeft(options, infoWidth) {
    return {
        height: '100%',
        width: infoWidth,
    };
}
function getInnerInfoStyle(placement, options, infoHeight, infoWidth) {
    var commonStyles = tslib_1.__assign(tslib_1.__assign({}, ((options.imageInfoType ===
        pro_gallery_lib_1.GALLERY_CONSTS.infoType.SEPARATED_BACKGROUND ||
        options.imageInfoType === pro_gallery_lib_1.GALLERY_CONSTS.infoType.ATTACHED_BACKGROUND) &&
        options.textBoxFillColor &&
        options.textBoxFillColor.value && {
        backgroundColor: options.textBoxFillColor.value,
    })), { overflow: 'hidden', boxSizing: 'border-box' });
    var infoAboveOrBelow = pro_gallery_lib_1.GALLERY_CONSTS.hasExternalVerticalPlacement(placement);
    var infoRightOrLeft = pro_gallery_lib_1.GALLERY_CONSTS.hasExternalHorizontalPlacement(placement);
    return tslib_1.__assign(tslib_1.__assign(tslib_1.__assign({}, commonStyles), (infoAboveOrBelow &&
        getInnerInfoStylesAboveOrBelow(options, infoHeight))), (infoRightOrLeft && getInnerInfoStylesRightOrLeft(options, infoWidth)));
}
exports.getInnerInfoStyle = getInnerInfoStyle;
