// pro-gallery-lib/dist/esm/settings/options/videoPlay.js
import { INPUT_TYPES } from '../utils/constants';
import { default as GALLERY_CONSTS } from '../../common/constants';
import { createOptions } from '../utils/utils';
import { default as layoutParams_navigationArrows_mouseCursorContainerMaxWidth } from './layoutParams_navigationArrows_mouseCursorContainerMaxWidth';
export default {
    title: 'Play Video On',
    description: "Choose when video items will start playing: On hover, on click or autoplay",
    isRelevant: function (options, option) {
        // specific isRelevant function
        var isVideoPlayAbleToPlay = function (options) {
            return !(options.layoutParams.navigationArrows.mouseCursorContainerMaxWidth ===
                100 &&
                layoutParams_navigationArrows_mouseCursorContainerMaxWidth.isRelevant(options));
        };
        // Distribution of the specific isRelevant functions
        var videoPlayOptions = {
            AUTO: function () {
                return true;
            },
            HOVER: isVideoPlayAbleToPlay,
            CLICK: isVideoPlayAbleToPlay,
        };
        // specific option isRelevant : general video play isRelevant (AUTO always true)
        return option
            ? videoPlayOptions[option](options)
            : Object.values(videoPlayOptions).some(function (val) { return val(options); });
    },
    isRelevantDescription: 'in case of HOVER or ON_CLICK - set "Arrows Position" to anything but "Mouse Cursor" (ON_GALLERY or OUTSIDE_GALLERY)',
    options: createOptions('videoPlay'),
    type: INPUT_TYPES.OPTIONS,
    default: GALLERY_CONSTS.videoPlay.HOVER,
};
