// pro-gallery/dist/cjs/components/gallery/typeValidator/typeErrorsUI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importDefault(require("react"));
function default_1(typeErrors) {
    window.typeErrors = typeErrors;
    return (react_1.default.createElement("div", null,
        react_1.default.createElement("ol", null, typeErrors.map(function (errObject, i) { return (react_1.default.createElement(Error, { errObject: errObject, key: i })); })),
        JSON.stringify(typeErrors, null, 4)));
}
exports.default = default_1;
function Error(_a) {
    var errObject = _a.errObject;
    return (react_1.default.createElement("li", null,
        react_1.default.createElement(ErrorDescription, { errObject: errObject })));
}
var ErrorDescription = /** @class */ (function (_super) {
    tslib_1.__extends(ErrorDescription, _super);
    function ErrorDescription(props) {
        var _this = _super.call(this, props) || this;
        _this.state = {
            showFull: false,
        };
        _this.toggle = _this.toggle.bind(_this); // PUKE
        return _this;
    }
    ErrorDescription.prototype.render = function () {
        var CompToRender = this.state.showFull ? FullError : Text;
        return (react_1.default.createElement("a", { onClick: this.toggle },
            react_1.default.createElement(CompToRender, { errObject: this.props.errObject })));
    };
    ErrorDescription.prototype.toggle = function () {
        this.setState({ showFull: !this.state.showFull });
    };
    return ErrorDescription;
}(react_1.default.Component));
function FullError(_a) {
    var errObject = _a.errObject;
    return (react_1.default.createElement("ul", null, Object.entries(errObject).map(function (entry, i) { return (react_1.default.createElement(Item, { entry: entry, key: i })); })));
}
function Item(_a) {
    var entry = _a.entry;
    var key = entry[0];
    var value = entry[1];
    var valueText = typeof value === 'string' ? value : JSON.stringify(value, null, 4);
    return react_1.default.createElement("li", null, key + ": " + valueText);
}
function Text(_a) {
    var errObject = _a.errObject;
    return errObject.message;
}
