// pro-gallery-lib/dist/cjs/common/window/viewModeWrapper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isDeviceTypeTouch = exports.isDeviceTypeDesktop = exports.isDeviceTypeTablet = exports.isDeviceTypeMobile = exports.isSEOMode = exports.isPreviewMode = exports.isEditMode = exports.isSiteMode = exports.viewModeWrapper = void 0;
var tslib_1 = require("tslib");
var viewMode_1 = tslib_1.__importDefault(require("../constants/viewMode"));
var deviceType_1 = tslib_1.__importDefault(require("../constants/deviceType"));
var ViewModeWrapper = /** @class */ (function () {
    function ViewModeWrapper() {
        this.setViewMode = this.setViewMode.bind(this);
        this.isSiteMode = this.isSiteMode.bind(this);
        this.isEditMode = this.isEditMode.bind(this);
        this.isPreviewMode = this.isPreviewMode.bind(this);
        this.isSEOMode = this.isSEOMode.bind(this);
        this.setDeviceType = this.setDeviceType.bind(this);
        this.isDeviceTypeDesktop = this.isDeviceTypeDesktop.bind(this);
        this.isDeviceTypeMobile = this.isDeviceTypeMobile.bind(this);
        this.isDeviceTypeTablet = this.isDeviceTypeTablet.bind(this);
        this.isDeviceTypeTouch = this.isDeviceTypeTouch.bind(this);
        this._viewMode = viewMode_1.default.SITE;
        this._deviceType = deviceType_1.default.DESKTOP;
    }
    ViewModeWrapper.prototype.setViewMode = function (viewMode) {
        this._viewMode = viewMode;
    };
    ViewModeWrapper.prototype.setDeviceType = function (forceVal) {
        this._deviceType = forceVal;
    };
    ViewModeWrapper.prototype.isDeviceTypeMobile = function () {
        return this._deviceType === deviceType_1.default.MOBILE;
    };
    ViewModeWrapper.prototype.isDeviceTypeTablet = function () {
        return this._deviceType === deviceType_1.default.TABLET;
    };
    ViewModeWrapper.prototype.isDeviceTypeDesktop = function () {
        return this._deviceType === deviceType_1.default.DESKTOP;
    };
    ViewModeWrapper.prototype.isDeviceTypeTouch = function () {
        return this.isMobile() || this.isTablet();
    };
    ViewModeWrapper.prototype.isSiteMode = function () {
        return this._viewMode === viewMode_1.default.SITE;
    };
    ViewModeWrapper.prototype.isEditMode = function () {
        return this._viewMode === viewMode_1.default.EDIT;
    };
    ViewModeWrapper.prototype.isPreviewMode = function () {
        return this._viewMode === viewMode_1.default.PREVIEW;
    };
    ViewModeWrapper.prototype.isSEOMode = function () {
        return this._viewMode === viewMode_1.default.SEO;
    };
    return ViewModeWrapper;
}());
var viewModeWrapper = new ViewModeWrapper();
exports.viewModeWrapper = viewModeWrapper;
var isSiteMode = viewModeWrapper.isSiteMode;
exports.isSiteMode = isSiteMode;
var isEditMode = viewModeWrapper.isEditMode;
exports.isEditMode = isEditMode;
var isPreviewMode = viewModeWrapper.isPreviewMode;
exports.isPreviewMode = isPreviewMode;
var isSEOMode = viewModeWrapper.isSEOMode;
exports.isSEOMode = isSEOMode;
var isDeviceTypeMobile = viewModeWrapper.isDeviceTypeMobile;
exports.isDeviceTypeMobile = isDeviceTypeMobile;
var isDeviceTypeTablet = viewModeWrapper.isDeviceTypeTablet;
exports.isDeviceTypeTablet = isDeviceTypeTablet;
var isDeviceTypeDesktop = viewModeWrapper.isDeviceTypeDesktop;
exports.isDeviceTypeDesktop = isDeviceTypeDesktop;
var isDeviceTypeTouch = viewModeWrapper.isDeviceTypeTouch;
exports.isDeviceTypeTouch = isDeviceTypeTouch;
