// pro-gallery/dist/cjs/components/gallery/proGallery/navigationArrows.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArrowsContainer = exports.ArrowButtonWithCursorController = exports.ArrowButton = exports.NavigationArrows = void 0;
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importDefault(require("react"));
var ReactDOM = tslib_1.__importStar(require("react-dom"));
var pro_gallery_lib_1 = require("pro-gallery-lib");
var mouseCursorPosition_1 = require("../../helpers/mouseCursorPosition");
var navigationArrowUtils_1 = require("../../helpers/navigationArrowUtils");
function NavigationArrows(_a) {
    var _b;
    var options = _a.options, hideLeftArrow = _a.hideLeftArrow, hideRightArrow = _a.hideRightArrow, container = _a.container, customNavArrowsRenderer = _a.customNavArrowsRenderer, next = _a.next, id = _a.id;
    var isRTL = options.isRTL, scrollDirection = options.scrollDirection, imageMargin = options.imageMargin, arrowsPadding = options.arrowsPadding, arrowsPosition = options.arrowsPosition, arrowsVerticalPosition = options.arrowsVerticalPosition, layoutParams = options.layoutParams, titlePlacement = options.titlePlacement, textBoxHeight = options.textBoxHeight, arrowsColor = options.arrowsColor, arrowsSize = options.arrowsSize;
    var _c = layoutParams.navigationArrows, _d = _c.container, type = _d.type, backgroundColor = _d.backgroundColor, borderRadius = _d.borderRadius, mouseCursorContainerMaxWidth = _c.mouseCursorContainerMaxWidth;
    var _e = navigationArrowUtils_1.getArrowsRenderData({
        customNavArrowsRenderer: customNavArrowsRenderer,
        arrowsColor: arrowsColor,
        arrowsSize: arrowsSize,
        arrowsType: layoutParams.navigationArrows.type,
        containerStyleType: type,
    }), renderArrowSvg = _e.arrowRenderer, navArrowsContainerWidth = _e.navArrowsContainerWidth, navArrowsContainerHeight = _e.navArrowsContainerHeight;
    var mouseCursorEnabled = arrowsPosition === pro_gallery_lib_1.GALLERY_CONSTS.arrowsPosition.MOUSE_CURSOR;
    var galleryHeight = container.galleryHeight;
    var galleryWidth = container.galleryWidth;
    var infoHeight = textBoxHeight;
    var imageHeight = galleryHeight - infoHeight;
    // the nav arrows parent container top edge is imageMargin/2 ABOVE the actual view, that calculates the middle point of gallery
    var galleryVerticalCenter = "50% + " + imageMargin / 4 + "px";
    // Determines the direction fix, the direction in which we move the nav arrows 'vertical position fix' pixels
    var directionFix;
    if (pro_gallery_lib_1.GALLERY_CONSTS.hasExternalAbovePlacement(titlePlacement)) {
        directionFix = -1;
    }
    else if (pro_gallery_lib_1.GALLERY_CONSTS.hasExternalBelowPlacement(titlePlacement)) {
        directionFix = 1;
    }
    else {
        // if we got here, we should be ITEM_CENTER, taken care of in layoutHelper.js
    }
    var verticalPositionFix = (_b = {},
        _b[pro_gallery_lib_1.GALLERY_CONSTS.arrowsVerticalPosition.ITEM_CENTER] = 0,
        _b[pro_gallery_lib_1.GALLERY_CONSTS.arrowsVerticalPosition.IMAGE_CENTER] = infoHeight * directionFix,
        _b[pro_gallery_lib_1.GALLERY_CONSTS.arrowsVerticalPosition.INFO_CENTER] = -imageHeight * directionFix,
        _b)[arrowsVerticalPosition];
    var arrowBoxStyle = navigationArrowUtils_1.getArrowBoxStyle({
        type: type,
        backgroundColor: backgroundColor,
        borderRadius: borderRadius,
    });
    var containerStyle = mouseCursorEnabled
        ? {
            width: galleryWidth + "px",
            maxWidth: mouseCursorContainerMaxWidth + "%",
            height: galleryHeight + "px",
            padding: 0,
            top: 0,
            flex: 1,
        }
        : tslib_1.__assign({ width: navArrowsContainerWidth + "px", height: navArrowsContainerHeight + "px", padding: 0, top: "calc(" + galleryVerticalCenter + " - " + navArrowsContainerHeight / 2 + "px - \n        " + verticalPositionFix / 2 + "px)" }, arrowBoxStyle);
    var arrowsPos = scrollDirection === pro_gallery_lib_1.GALLERY_CONSTS.scrollDirection.HORIZONTAL &&
        arrowsPosition === pro_gallery_lib_1.GALLERY_CONSTS.arrowsPosition.OUTSIDE_GALLERY
        ? "-" + (20 + navArrowsContainerWidth) + "px"
        : imageMargin / 2 + (arrowsPadding ? arrowsPadding : 0) + "px";
    // imageMargin effect the margin of the main div ('pro-gallery-parent-container') that SlideshowView is rendering, so the arrows should be places accordingly
    // arrowsPadding relevant only for arrowsPosition.ON_GALLERY
    var prevContainerStyle = { left: mouseCursorEnabled ? 0 : arrowsPos };
    var nextContainerStyle = { right: mouseCursorEnabled ? 0 : arrowsPos };
    var useDropShadow = type === pro_gallery_lib_1.GALLERY_CONSTS.arrowsContainerStyleType.SHADOW;
    var arrowsBaseClasses = [
        'nav-arrows-container',
        useDropShadow ? 'drop-shadow' : '',
        pro_gallery_lib_1.utils.isMobile() ? ' pro-gallery-mobile-indicator' : '',
        mouseCursorEnabled ? 'follow-mouse-cursor' : '',
    ];
    var navigationArrowPortalId = "arrow-portal-container-" + id;
    var ArrowRenderHandler = mouseCursorEnabled
        ? ArrowButtonWithCursorController
        : ArrowButton;
    var renderArrow = function (directionIsLeft) {
        return (react_1.default.createElement(ArrowRenderHandler, tslib_1.__assign({}, {
            renderArrowSvg: renderArrowSvg,
            next: next,
            directionIsLeft: directionIsLeft,
            arrowsBaseClasses: arrowsBaseClasses,
            tabIndex: pro_gallery_lib_1.utils.getTabIndex.bind(pro_gallery_lib_1.utils),
            containerStyle: containerStyle,
            prevContainerStyle: prevContainerStyle,
            nextContainerStyle: nextContainerStyle,
            isRTL: isRTL,
            hideLeftArrow: hideLeftArrow,
            arrowBoxStyle: arrowBoxStyle,
            navArrowsContainerWidth: navArrowsContainerWidth,
            navArrowsContainerHeight: navArrowsContainerHeight,
            navigationArrowPortalId: navigationArrowPortalId,
        })));
    };
    return (react_1.default.createElement(ArrowsContainer, { hideLeftArrow: hideLeftArrow, hideRightArrow: hideRightArrow, mouseCursorEnabled: mouseCursorEnabled, isRTL: isRTL, navigationArrowPortalId: navigationArrowPortalId },
        hideLeftArrow ? null : renderArrow(true),
        hideRightArrow ? null : renderArrow(false)));
}
exports.NavigationArrows = NavigationArrows;
function ArrowButton(_a) {
    var _b;
    var cursor = _a.cursor, arrowsBaseClasses = _a.arrowsBaseClasses, containerStyle = _a.containerStyle, prevContainerStyle = _a.prevContainerStyle, nextContainerStyle = _a.nextContainerStyle, isRTL = _a.isRTL, directionIsLeft = _a.directionIsLeft, renderArrowSvg = _a.renderArrowSvg, next = _a.next, tabIndex = _a.tabIndex, arrowBoxStyle = _a.arrowBoxStyle, navArrowsContainerWidth = _a.navArrowsContainerWidth, navArrowsContainerHeight = _a.navArrowsContainerHeight, navigationArrowPortalId = _a.navigationArrowPortalId;
    var isNext = (directionIsLeft && isRTL) || (!directionIsLeft && !isRTL);
    var buttonProps = (_b = {
            className: arrowsBaseClasses.join(' '),
            onClick: function () { return next({ direction: directionIsLeft ? -1 : 1 }); }
        },
        _b['aria-label'] = (isNext ? 'Next' : 'Previous') + " Item",
        _b.tabIndex = tabIndex(isNext ? 'slideshowNext' : 'slideshowPrev'),
        _b.key = !isNext ? 'nav-arrow-back' : 'nav-arrow-next',
        _b['data-hook'] = !isNext ? 'nav-arrow-back' : 'nav-arrow-next',
        _b.style = tslib_1.__assign(tslib_1.__assign({}, containerStyle), (directionIsLeft ? prevContainerStyle : nextContainerStyle)),
        _b);
    if (cursor) {
        var containerRef_1 = cursor.containerRef, isMouseEnter = cursor.isMouseEnter, position = cursor.position;
        return (react_1.default.createElement("button", tslib_1.__assign({ ref: function (ref) { return (containerRef_1.current = ref); } }, buttonProps), isMouseEnter &&
            ReactDOM.createPortal(react_1.default.createElement("span", { style: tslib_1.__assign(tslib_1.__assign({ top: position.y - navArrowsContainerHeight / 2, left: position.x - navArrowsContainerWidth / 2 }, arrowBoxStyle), { width: navArrowsContainerWidth, height: navArrowsContainerHeight }) }, renderArrowSvg(directionIsLeft ? 'left' : 'right')), window.document.getElementById(navigationArrowPortalId))));
    }
    return (react_1.default.createElement("button", tslib_1.__assign({}, buttonProps), renderArrowSvg(directionIsLeft ? 'left' : 'right')));
}
exports.ArrowButton = ArrowButton;
function ArrowButtonWithCursorController(props) {
    return (react_1.default.createElement(mouseCursorPosition_1.CursorController, null, function (_a) {
        var containerRef = _a.containerRef, position = _a.position, isMouseEnter = _a.isMouseEnter;
        return (react_1.default.createElement(ArrowButton, tslib_1.__assign({ cursor: {
                containerRef: containerRef,
                position: position,
                isMouseEnter: isMouseEnter,
            } }, props)));
    }));
}
exports.ArrowButtonWithCursorController = ArrowButtonWithCursorController;
function ArrowsContainer(_a) {
    var hideRightArrow = _a.hideRightArrow, hideLeftArrow = _a.hideLeftArrow, mouseCursorEnabled = _a.mouseCursorEnabled, isRTL = _a.isRTL, children = _a.children, navigationArrowPortalId = _a.navigationArrowPortalId;
    if (mouseCursorEnabled) {
        var styleForMouseCursor = {
            justifyContent: hideLeftArrow
                ? 'flex-end'
                : hideRightArrow
                    ? 'flex-start'
                    : 'space-between',
            flexDirection: isRTL ? 'row-reverse' : 'row',
        };
        return (react_1.default.createElement("div", { className: "mouse-cursor", style: tslib_1.__assign({}, styleForMouseCursor) },
            react_1.default.createElement("div", { id: navigationArrowPortalId, className: "arrow-portal-container" }),
            children));
    }
    return react_1.default.createElement(react_1.default.Fragment, null, children);
}
exports.ArrowsContainer = ArrowsContainer;
