// pro-gallery/dist/cjs/components/gallery/proGallery/proGallery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
require("../../../common/utils/polyfills");
var react_1 = tslib_1.__importDefault(require("react"));
var pro_gallery_lib_1 = require("pro-gallery-lib");
var galleryContainer_js_1 = tslib_1.__importDefault(require("./galleryContainer.js"));
require("../../../versionLogger");
var ProGallery = /** @class */ (function (_super) {
    tslib_1.__extends(ProGallery, _super);
    function ProGallery(props) {
        var _this = _super.call(this) || this;
        _this.init(props);
        if (pro_gallery_lib_1.utils.isLocal() && !pro_gallery_lib_1.utils.isTest()) {
            console.log('PRO GALLERY DEV');
        }
        return _this;
    }
    ProGallery.prototype.init = function (props) {
        if (typeof props.viewMode !== 'undefined') {
            pro_gallery_lib_1.viewModeWrapper.setViewMode(props.viewMode);
        }
        if (typeof props.deviceType !== 'undefined') {
            pro_gallery_lib_1.viewModeWrapper.setDeviceType(props.deviceType);
        }
    };
    ProGallery.prototype.UNSAFE_componentWillReceiveProps = function (nextProps) {
        if (this.props.viewMode !== nextProps.viewMode) {
            pro_gallery_lib_1.utils.dumpCache();
            pro_gallery_lib_1.viewModeWrapper.setViewMode(nextProps.viewMode);
        }
        if (this.props.deviceType !== nextProps.deviceType) {
            pro_gallery_lib_1.utils.dumpCache();
            pro_gallery_lib_1.viewModeWrapper.setDeviceType(nextProps.deviceType);
        }
    };
    ProGallery.prototype.renderProps = function () {
        var _a;
        return tslib_1.__assign(tslib_1.__assign({}, this.props), { id: this.props.id, items: this.props.items || [], settings: this.props.settings || {}, offsetTop: this.props.offsetTop, proGalleryRegionLabel: this.props.proGalleryRegionLabel, isInDisplay: (_a = this.props.isInDisplay) !== null && _a !== void 0 ? _a : true });
    };
    ProGallery.prototype.containerProps = function () {
        return {
            id: "pro-gallery-" + this.props.id,
            className: 'pro-gallery',
        };
    };
    ProGallery.prototype.render = function () {
        return (react_1.default.createElement("div", tslib_1.__assign({}, this.containerProps()),
            react_1.default.createElement(galleryContainer_js_1.default, tslib_1.__assign({}, this.renderProps()))));
    };
    return ProGallery;
}(react_1.default.Component));
exports.default = ProGallery;
