// pro-gallery-lib/dist/cjs/core/presets/presets.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isInPreset = exports.getLayoutName = exports.NEW_PRESETS = exports.addPresetOptions = void 0;
var tslib_1 = require("tslib");
var layout_1 = tslib_1.__importDefault(require("../../common/constants/layout"));
//#region Imports Layouts
var alternateGallery_1 = require("./alternateGallery");
var bricksGallery_1 = require("./bricksGallery");
var collageGallery_1 = require("./collageGallery");
var fullsizeGallery_1 = require("./fullsizeGallery");
var columnGallery_1 = require("./columnGallery");
var gridGallery_1 = require("./gridGallery");
var magicGallery_1 = require("./magicGallery");
var masonryGallery_1 = require("./masonryGallery");
var mixGallery_1 = require("./mixGallery");
var panoramaGallery_1 = require("./panoramaGallery");
var sliderGallery_1 = require("./sliderGallery");
var slideshowGallery_1 = require("./slideshowGallery");
var thumbnailGallery_1 = require("./thumbnailGallery");
var emptyGallery_1 = require("./emptyGallery");
var designedPresetGallery_1 = require("./designedPresetGallery");
var optionsConverter_1 = require("../helpers/optionsConverter");
var optionsUtils_1 = require("../helpers/optionsUtils");
//#endregion Imports
var addPresetOptions = function (options) {
    var galleryType = options.galleryType;
    var galleryLayoutV1 = options.galleryType;
    var galleryLayoutV2 = options.galleryLayout;
    if (galleryLayoutV1 !== undefined && galleryLayoutV2 === undefined) {
        // legacy layouts - only if galleyrType parameter is specifically defined (i.e. layout had changed)
        switch (galleryType) {
            case '1': // horizontal collage
                return collageGallery_1.createOptions(options);
            case '2': // grid
                return gridGallery_1.createOptions(options);
            case '3': // vertical masonry
                return masonryGallery_1.createOptions(options);
            case '4': // horizontal masonry
                return masonryGallery_1.createOptions(options);
            case '5': // one column
                return panoramaGallery_1.createOptions(options);
            case '6': // one row
                return columnGallery_1.createOptions(options);
            case '7': // slideshow
                return slideshowGallery_1.createOptions(options);
            case '13': // slideshow
                return slideshowGallery_1.createOptions(options);
            case '0': // vertical collage
            default:
                return collageGallery_1.createOptions(options);
        }
    }
    else {
        // new layouts
        switch (options.galleryLayout) {
            case layout_1.default.MASONRY:
                return masonryGallery_1.createOptions(options);
            case layout_1.default.GRID:
                return gridGallery_1.createOptions(options);
            case layout_1.default.THUMBNAIL:
                return thumbnailGallery_1.createOptions(options);
            case layout_1.default.SLIDER:
                return sliderGallery_1.createOptions(options);
            case layout_1.default.SLIDESHOW:
                return slideshowGallery_1.createOptions(options);
            case layout_1.default.PANORAMA:
                return panoramaGallery_1.createOptions(options);
            case layout_1.default.COLUMN:
                return columnGallery_1.createOptions(options);
            case layout_1.default.MAGIC:
                return magicGallery_1.createOptions(options);
            case layout_1.default.FULLSIZE:
                return fullsizeGallery_1.createOptions(options);
            case layout_1.default.BRICKS:
                return bricksGallery_1.createOptions(options);
            case layout_1.default.MIX:
                return mixGallery_1.createOptions(options);
            case layout_1.default.ALTERNATE:
                return alternateGallery_1.createOptions(options);
            case layout_1.default.EMPTY:
                return emptyGallery_1.createOptions(options);
            case layout_1.default.JSON_FIXED:
                return designedPresetGallery_1.createOptions(options);
            case layout_1.default.COLLAGE:
            default:
                return collageGallery_1.createOptions(options);
        }
    }
};
exports.addPresetOptions = addPresetOptions;
var NEW_PRESETS = {
    alternate: alternateGallery_1.fixedOptions,
    bricks: bricksGallery_1.fixedOptions,
    collage: collageGallery_1.fixedOptions,
    fullsize: fullsizeGallery_1.fixedOptions,
    column: columnGallery_1.fixedOptions,
    grid: gridGallery_1.fixedOptions,
    magic: magicGallery_1.fixedOptions,
    masonry: masonryGallery_1.fixedOptions,
    mix: mixGallery_1.fixedOptions,
    panorama: panoramaGallery_1.fixedOptions,
    slider: sliderGallery_1.fixedOptions,
    slideshow: slideshowGallery_1.fixedOptions,
    thumbnails: thumbnailGallery_1.fixedOptions,
    empty: emptyGallery_1.fixedOptions,
};
exports.NEW_PRESETS = NEW_PRESETS;
var getLayoutName = function (galleryLayout) {
    var galleyLayoutList = [
        'empty',
        'collage',
        'masonry',
        'grid',
        'thumbnails',
        'slider',
        'slideshow',
        'panorama',
        'column',
        'magic',
        'fullsize',
        'bricks',
        'mix',
        'alternate',
    ];
    return galleyLayoutList[galleryLayout + 1];
};
exports.getLayoutName = getLayoutName;
// returns true if the given param is in the current layout preset
var isInPreset = function (galleryLayout, paramToCheck) {
    var layoutName = getLayoutName(galleryLayout) || 'empty'; // empty for when there is no layout given
    return Object.keys(optionsConverter_1.addMigratedOptions(optionsUtils_1.flattenObject(NEW_PRESETS[layoutName]))).includes(paramToCheck);
};
exports.isInPreset = isInPreset;
