// pro-gallery-lib/dist/cjs/common/constants/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var arrowsPosition_1 = tslib_1.__importDefault(require("./arrowsPosition"));
var arrowsVerticalPosition_1 = tslib_1.__importDefault(require("./arrowsVerticalPosition"));
var cubeType_1 = tslib_1.__importDefault(require("./cubeType"));
var dimensions_1 = tslib_1.__importDefault(require("./dimensions"));
var events_1 = tslib_1.__importDefault(require("./events"));
var deviceType_1 = tslib_1.__importDefault(require("./deviceType"));
var gallerySizeType_1 = tslib_1.__importDefault(require("./gallerySizeType"));
var galleryTextAlign_js_1 = tslib_1.__importDefault(require("./galleryTextAlign.js"));
var gridStyle_1 = tslib_1.__importDefault(require("./gridStyle"));
var cubeFitPosition_1 = tslib_1.__importDefault(require("./cubeFitPosition"));
var groupTypes_1 = tslib_1.__importDefault(require("./groupTypes"));
var imageHoverAnimations_1 = tslib_1.__importDefault(require("./imageHoverAnimations"));
var infoBehaviourOnHover_1 = tslib_1.__importDefault(require("./infoBehaviourOnHover"));
var infoType_1 = tslib_1.__importDefault(require("./infoType"));
var isVertical_1 = tslib_1.__importDefault(require("./isVertical"));
var itemClick_1 = tslib_1.__importDefault(require("./itemClick"));
var layout_1 = tslib_1.__importStar(require("./layout"));
var layoutDirection_1 = tslib_1.__importDefault(require("./layoutDirection"));
var loadingMode_1 = tslib_1.__importDefault(require("./loadingMode"));
var loadingWithColorMode_1 = tslib_1.__importDefault(require("./loadingWithColorMode"));
var loadMoreAmount_1 = tslib_1.__importDefault(require("./loadMoreAmount"));
var mobileSwipeAnimations_1 = tslib_1.__importDefault(require("./mobileSwipeAnimations"));
var overlayAnimations_1 = tslib_1.__importDefault(require("./overlayAnimations"));
var placements_1 = tslib_1.__importStar(require("./placements"));
var resizeMethods_1 = tslib_1.__importDefault(require("./resizeMethods"));
var scrollAnimations_1 = tslib_1.__importDefault(require("./scrollAnimations"));
var slideAnimations_1 = tslib_1.__importDefault(require("./slideAnimations"));
var scrollDirection_1 = tslib_1.__importDefault(require("./scrollDirection"));
var socialNetworks_1 = tslib_1.__importDefault(require("./socialNetworks"));
var textBoxWidthCalculationOptions_1 = tslib_1.__importDefault(require("./textBoxWidthCalculationOptions"));
var thumbnailsAlignment_1 = tslib_1.__importDefault(require("./thumbnailsAlignment"));
var thumbnailsPosition_1 = tslib_1.__importDefault(require("./thumbnailsPosition"));
var urlTypes_1 = require("./urlTypes");
var videoPlay_1 = tslib_1.__importDefault(require("./videoPlay"));
var viewMode_1 = tslib_1.__importDefault(require("./viewMode"));
var imagePlacementAnimations_1 = tslib_1.__importDefault(require("./imagePlacementAnimations"));
var slideTransition_1 = tslib_1.__importDefault(require("./slideTransition"));
var overlayPositions_1 = tslib_1.__importDefault(require("./overlayPositions"));
var overlaySizeType_1 = tslib_1.__importDefault(require("./overlaySizeType"));
var autoSlideshowTypes_1 = tslib_1.__importDefault(require("./autoSlideshowTypes"));
var arrowsType_1 = tslib_1.__importDefault(require("./arrowsType"));
var arrowsContainerStyleType_1 = tslib_1.__importDefault(require("./arrowsContainerStyleType"));
var itemResolutionMode_1 = tslib_1.__importDefault(require("./itemResolutionMode"));
var autoSlideBehaviour_1 = tslib_1.__importDefault(require("./autoSlideBehaviour"));
var secondaryMediaTrigger_1 = tslib_1.__importDefault(require("./secondaryMediaTrigger"));
var secondaryMediaBehaviour_1 = tslib_1.__importDefault(require("./secondaryMediaBehaviour"));
exports.default = {
    arrowsPosition: arrowsPosition_1.default,
    itemResolutionMode: itemResolutionMode_1.default,
    arrowsVerticalPosition: arrowsVerticalPosition_1.default,
    cubeType: cubeType_1.default,
    dimensions: dimensions_1.default,
    events: events_1.default,
    deviceType: deviceType_1.default,
    gallerySizeType: gallerySizeType_1.default,
    galleryTextAlign: galleryTextAlign_js_1.default,
    gridStyle: gridStyle_1.default,
    cubeFitPosition: cubeFitPosition_1.default,
    groupTypes: groupTypes_1.default,
    imageHoverAnimations: imageHoverAnimations_1.default,
    infoBehaviourOnHover: infoBehaviourOnHover_1.default,
    infoType: infoType_1.default,
    isVertical: isVertical_1.default,
    itemClick: itemClick_1.default,
    layout: layout_1.default,
    isLayout: layout_1.isLayout,
    layoutDirection: layoutDirection_1.default,
    loadingMode: loadingMode_1.default,
    loadingWithColorMode: loadingWithColorMode_1.default,
    loadMoreAmount: loadMoreAmount_1.default,
    mobileSwipeAnimations: mobileSwipeAnimations_1.default,
    overlayAnimations: overlayAnimations_1.default,
    placements: placements_1.default,
    hasExternalAbovePlacement: placements_1.hasExternalAbovePlacement,
    hasExternalBelowPlacement: placements_1.hasExternalBelowPlacement,
    hasHoverPlacement: placements_1.hasHoverPlacement,
    hasExternalRightPlacement: placements_1.hasExternalRightPlacement,
    hasExternalLeftPlacement: placements_1.hasExternalLeftPlacement,
    hasExternalVerticalPlacement: placements_1.hasExternalVerticalPlacement,
    hasExternalHorizontalPlacement: placements_1.hasExternalHorizontalPlacement,
    isExternalAbovePlacement: placements_1.isExternalAbovePlacement,
    isExternalBelowPlacement: placements_1.isExternalBelowPlacement,
    isHoverPlacement: placements_1.isHoverPlacement,
    isExternalRightPlacement: placements_1.isExternalRightPlacement,
    isExternalLeftPlacement: placements_1.isExternalLeftPlacement,
    isExternalVerticalPlacement: placements_1.isExternalVerticalPlacement,
    isExternalHorizontalPlacement: placements_1.isExternalHorizontalPlacement,
    isConstantVerticalPlacement: placements_1.isConstantVerticalPlacement,
    resizeMethods: resizeMethods_1.default,
    scrollAnimations: scrollAnimations_1.default,
    slideAnimations: slideAnimations_1.default,
    scrollDirection: scrollDirection_1.default,
    socialNetworks: socialNetworks_1.default,
    textBoxWidthCalculationOptions: textBoxWidthCalculationOptions_1.default,
    thumbnailsAlignment: thumbnailsAlignment_1.default,
    thumbnailsPosition: thumbnailsPosition_1.default,
    urlSizes: urlTypes_1.URL_SIZES,
    urlTypes: urlTypes_1.URL_TYPES,
    videoPlay: videoPlay_1.default,
    viewMode: viewMode_1.default,
    imagePlacementAnimations: imagePlacementAnimations_1.default,
    slideTransition: slideTransition_1.default,
    overlayPositions: overlayPositions_1.default,
    overlaySizeType: overlaySizeType_1.default,
    autoSlideshowTypes: autoSlideshowTypes_1.default,
    arrowsType: arrowsType_1.default,
    arrowsContainerStyleType: arrowsContainerStyleType_1.default,
    autoSlideBehaviour: autoSlideBehaviour_1.default,
    secondaryMediaTrigger: secondaryMediaTrigger_1.default,
    secondaryMediaBehaviour: secondaryMediaBehaviour_1.default,
};
