// pro-gallery-lib/dist/cjs/common/constants/placements.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isConstantVerticalPlacement = exports.isExternalHorizontalPlacement = exports.isExternalVerticalPlacement = exports.isExternalLeftPlacement = exports.isExternalRightPlacement = exports.isHoverPlacement = exports.isExternalBelowPlacement = exports.isExternalAbovePlacement = exports.hasExternalHorizontalPlacement = exports.hasExternalVerticalPlacement = exports.hasExternalLeftPlacement = exports.hasExternalRightPlacement = exports.hasHoverPlacement = exports.hasExternalBelowPlacement = exports.hasExternalAbovePlacement = void 0;
var PLACEMENTS = {
    SHOW_ON_HOVER: 'SHOW_ON_HOVER',
    SHOW_BELOW: 'SHOW_BELOW',
    SHOW_ABOVE: 'SHOW_ABOVE',
    SHOW_ON_THE_RIGHT: 'SHOW_ON_THE_RIGHT',
    SHOW_ON_THE_LEFT: 'SHOW_ON_THE_LEFT',
    ALTERNATE_HORIZONTAL: 'ALTERNATE_HORIZONTAL',
    ALTERNATE_VERTICAL: 'ALTERNATE_VERTICAL',
};
var NEW_PLACEMENTS = {
    OVERLAY: 'OVERLAY',
    ABOVE: 'ABOVE',
    BELOW: 'BELOW',
    LEFT: 'LEFT',
    RIGHT: 'RIGHT',
    ALTERNATE_HORIZONTALLY: 'ALTERNATE_HORIZONTALLY',
    ALTERNATE_VERTICALLY: 'ALTERNATE_VERTICALLY',
};
var hasHoverPlacement = function (placement) {
    return String(placement).includes(PLACEMENTS.SHOW_ON_HOVER) ||
        String(placement).includes(NEW_PLACEMENTS.OVERLAY);
};
exports.hasHoverPlacement = hasHoverPlacement;
var hasExternalAbovePlacement = function (placement, idx) {
    return String(placement).includes(PLACEMENTS.SHOW_ABOVE) ||
        String(placement).includes(NEW_PLACEMENTS.ABOVE) ||
        (idx % 2 === 0 &&
            (String(placement).includes(PLACEMENTS.ALTERNATE_VERTICAL) ||
                String(placement).includes(NEW_PLACEMENTS.ALTERNATE_VERTICALLY)));
};
exports.hasExternalAbovePlacement = hasExternalAbovePlacement;
var hasExternalBelowPlacement = function (placement, idx) {
    return String(placement).includes(PLACEMENTS.SHOW_BELOW) ||
        String(placement).includes(NEW_PLACEMENTS.BELOW) ||
        (idx % 2 === 1 &&
            (String(placement).includes(PLACEMENTS.ALTERNATE_VERTICAL) ||
                String(placement).includes(NEW_PLACEMENTS.ALTERNATE_VERTICALLY)));
};
exports.hasExternalBelowPlacement = hasExternalBelowPlacement;
var hasExternalRightPlacement = function (placement, idx) {
    return String(placement).includes(PLACEMENTS.SHOW_ON_THE_RIGHT) ||
        String(placement).includes(NEW_PLACEMENTS.RIGHT) ||
        (idx % 2 === 0 &&
            String(placement).includes(PLACEMENTS.ALTERNATE_HORIZONTAL)) ||
        String(placement).includes(NEW_PLACEMENTS.ALTERNATE_HORIZONTALLY);
};
exports.hasExternalRightPlacement = hasExternalRightPlacement;
var hasExternalLeftPlacement = function (placement, idx) {
    return String(placement).includes(PLACEMENTS.SHOW_ON_THE_LEFT) ||
        (idx % 2 === 1 &&
            String(placement).includes(PLACEMENTS.ALTERNATE_HORIZONTAL)) ||
        String(placement).includes(NEW_PLACEMENTS.ALTERNATE_HORIZONTALLY);
};
exports.hasExternalLeftPlacement = hasExternalLeftPlacement;
var hasExternalVerticalPlacement = function (placement) {
    return hasExternalAbovePlacement(placement, 0) ||
        hasExternalBelowPlacement(placement, 1);
};
exports.hasExternalVerticalPlacement = hasExternalVerticalPlacement;
var hasExternalHorizontalPlacement = function (placement) {
    return hasExternalRightPlacement(placement, 0) ||
        hasExternalLeftPlacement(placement, 1);
};
exports.hasExternalHorizontalPlacement = hasExternalHorizontalPlacement;
var isExternalVerticalPlacement = function (placement) {
    return hasExternalVerticalPlacement(placement) &&
        !hasExternalHorizontalPlacement(placement) &&
        !hasHoverPlacement(placement);
};
exports.isExternalVerticalPlacement = isExternalVerticalPlacement;
var isExternalHorizontalPlacement = function (placement) {
    return hasExternalHorizontalPlacement(placement) &&
        !hasExternalVerticalPlacement(placement) &&
        !hasHoverPlacement(placement);
};
exports.isExternalHorizontalPlacement = isExternalHorizontalPlacement;
var isExternalAbovePlacement = function (placement) {
    return String(placement) === PLACEMENTS.SHOW_ABOVE ||
        String(placement) === NEW_PLACEMENTS.ABOVE;
};
exports.isExternalAbovePlacement = isExternalAbovePlacement;
var isExternalBelowPlacement = function (placement) {
    return String(placement) === PLACEMENTS.SHOW_BELOW ||
        String(placement) === NEW_PLACEMENTS.BELOW;
};
exports.isExternalBelowPlacement = isExternalBelowPlacement;
var isExternalAboveOrBelowSinglePlacement = function (placement) {
    return isExternalAbovePlacement(placement) || isExternalBelowPlacement(placement);
};
var isHoverPlacement = function (placement) {
    return String(placement) === PLACEMENTS.SHOW_ON_HOVER ||
        String(placement) === NEW_PLACEMENTS.OVERLAY;
};
exports.isHoverPlacement = isHoverPlacement;
var isExternalRightPlacement = function (placement) {
    return String(placement) === PLACEMENTS.SHOW_ON_THE_RIGHT ||
        String(placement) === NEW_PLACEMENTS.RIGHT;
};
exports.isExternalRightPlacement = isExternalRightPlacement;
var isExternalLeftPlacement = function (placement) {
    return String(placement) === PLACEMENTS.SHOW_ON_THE_LEFT ||
        String(placement) === NEW_PLACEMENTS.LEFT;
};
exports.isExternalLeftPlacement = isExternalLeftPlacement;
var isConstantVerticalPlacement = function (placement) {
    var placementArray = placement.split(',');
    var firstPlacement = placementArray.shift();
    return (isExternalAboveOrBelowSinglePlacement(firstPlacement) &&
        placementArray.every(function (placement) { return placement === firstPlacement; }));
};
exports.isConstantVerticalPlacement = isConstantVerticalPlacement;
exports.default = PLACEMENTS;
