// pro-gallery-lib/dist/cjs/common/defaultOptions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.populateWithDefaultOptions = void 0;
var tslib_1 = require("tslib");
var constants_1 = tslib_1.__importDefault(require("./constants"));
var coreOptions_1 = tslib_1.__importDefault(require("./coreOptions"));
var optionsUtils_1 = require("../core/helpers/optionsUtils");
var defaultOptions = optionsUtils_1.mergeNestedObjects(coreOptions_1.default, {
    layoutParams: {
        repeatingGroupTypes: '',
        navigationArrows: {
            type: constants_1.default.arrowsType.DEFAULT_ARROW,
            container: {
                type: constants_1.default.arrowsContainerStyleType.SHADOW,
                backgroundColor: 'rgba(0,0,0,0)',
                borderRadius: 0,
            },
            mouseCursorContainerMaxWidth: 100,
        },
        thumbnails: {
            position: constants_1.default.thumbnailsPosition.OUTSIDE_GALLERY,
        },
        structure: {
            galleryRatio: { value: 0, includeExternalInfo: false },
        },
    },
    behaviourParams: {
        item: {
            content: {
                magnificationValue: 2,
            },
            secondaryMedia: {
                trigger: constants_1.default.secondaryMediaTrigger.OFF,
                behaviour: constants_1.default.secondaryMediaBehaviour.APPEARS,
            },
        },
    },
    // adding
    galleryLayout: -1,
    gallerySizePx: 0,
    gallerySizeRatio: 0,
    gallerySizeType: constants_1.default.gallerySizeType.SMART,
    itemShadowOpacityAndColor: '',
    arrowsColor: '',
    textBoxBorderColor: '',
    allowContextMenu: false,
    showVideoPlayButton: false,
    // ----
    gallerySize: 30,
    cropOnlyFill: false,
    rotatingCropRatios: '',
    columnWidths: '',
    numberOfImagesPerCol: 1,
    groupsPerStrip: 0,
    imageMargin: 10,
    scatter: 0,
    rotatingScatter: '',
    placeGroupsLtr: false,
    enableInfiniteScroll: true,
    thumbnailSpacings: 4,
    enableScroll: true,
    scrollSnap: false,
    itemClick: constants_1.default.itemClick.NOTHING,
    slideAnimation: constants_1.default.slideAnimations.SCROLL,
    scrollDuration: 400,
    arrowsPosition: constants_1.default.arrowsPosition.ON_GALLERY,
    arrowsVerticalPosition: constants_1.default.arrowsVerticalPosition.ITEM_CENTER,
    arrowsPadding: 23,
    thumbnailSize: 120,
    magicLayoutSeed: 1,
    imagePlacementAnimation: constants_1.default.imagePlacementAnimations.NO_EFFECT,
    calculateTextBoxWidthMode: constants_1.default.textBoxWidthCalculationOptions.PERCENT,
    textBoxWidth: 200,
    textBoxWidthPercent: 50,
    loadMoreButtonText: '',
    videoSpeed: '1',
    showVideoControls: false,
    shouldIndexDirectShareLinkInSEO: true,
    slideTransition: constants_1.default.slideTransition.ease,
    useMaxDimensions: false,
    enableVideoPlaceholder: true,
    overlayPosition: constants_1.default.overlayPositions.LEFT,
    overlaySize: 100,
    overlaySizeType: constants_1.default.overlaySizeType.PERCENT,
    overlayPadding: 0,
    cubeFitPosition: constants_1.default.cubeFitPosition.MIDDLE,
    //migrated: keep here or it will still break users of this (no dependency on refactor)
    magnificationLevel: 2,
});
function populateWithDefaultOptions(options) {
    var flatDefault = optionsUtils_1.flattenObject(defaultOptions);
    var flatOptions = optionsUtils_1.flattenObject(options);
    var mergedOptions = Object.assign({}, flatDefault, flatOptions);
    Object.keys(mergedOptions).forEach(function (key) {
        if (typeof mergedOptions[key] === 'undefined') {
            mergedOptions[key] = defaultOptions[key];
        }
    });
    return optionsUtils_1.flatToNested(mergedOptions);
}
exports.populateWithDefaultOptions = populateWithDefaultOptions;
exports.default = defaultOptions;
