// pro-gallery-lib/dist/cjs/common/v4DefaultOptions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = tslib_1.__importDefault(require("./constants"));
var defaultOptions = {
    layoutParams: {
        crop: {
            ratios: [1],
            method: 'FILL',
            enable: false,
            enableSmartCrop: false,
            cropOnlyFill: false,
            alignment: 'CENTER',
        },
        structure: {
            galleryLayout: -1,
            scrollDirection: 'VERTICAL',
            gallerySpacing: 0,
            itemSpacing: 10,
            enableStreching: true,
            responsiveMode: 'FIT_TO_SCREEN',
            scatter: {
                randomScatter: 0,
                manualScatter: '',
            },
            layoutOrientation: 'HORIZONTAL',
            groupsOrder: 'BY_HEIGHT',
            numberOfGridRows: 1,
            numberOfColumns: 3,
            columnRatios: [],
        },
        groups: {
            groupByOrientation: true,
            numberOfGroupsPerRow: 0,
            density: 0.8,
            groupSize: 3,
            allowedGroupTypes: ['1', '2h', '2v', '3t', '3b', '3l', '3r'],
            repeatingGroupTypes: [],
        },
        thumbnails: {
            size: 120,
            spacing: 4,
            enable: false,
            alignment: 'BOTTOM',
        },
        navigationArrows: {
            enable: true,
            position: 'ON_GALLERY',
            padding: 23,
            size: 23,
            verticalAlignment: constants_1.default.arrowsVerticalPosition.ITEM_CENTER,
            type: constants_1.default.arrowsType.DEFAULT_ARROW,
            container: {
                type: constants_1.default.arrowsContainerStyleType.SHADOW,
                backgroundColor: 'rgba(0,0,0,0)',
                borderRadius: 0,
            },
        },
        targetItemSize: {
            unit: 'SMART',
            value: 30,
            minimum: 120,
        },
        info: {
            sizeUnits: constants_1.default.textBoxWidthCalculationOptions.PERCENT,
            width: 50,
            height: 200,
            spacing: 10,
            layout: constants_1.default.infoType.NO_BACKGROUND,
            // widthByPercent: 50, //I want this to be in the width just like we did the overlaySize
            placement: 'OVERLAY',
            border: {
                width: 0,
                radius: 0,
                color: '',
            },
        },
    },
    behaviourParams: {
        item: {
            clickAction: 'NOTHING',
            video: {
                playTrigger: 'HOVER',
                loop: true,
                volume: 0,
                enableControls: false,
                speed: 1,
                enablePlayButton: false,
                enablePlaceholder: true,
            },
            overlay: {
                hoveringBehaviour: constants_1.default.infoBehaviourOnHover.APPEARS,
                hoverAnimation: constants_1.default.overlayAnimations.NO_EFFECT,
                position: constants_1.default.overlayPositions.LEFT,
                size: 100,
                sizeUnits: constants_1.default.overlaySizeType.PERCENT,
                padding: 0,
            },
            content: {
                hoverAnimation: constants_1.default.imageHoverAnimations.NO_EFFECT,
                loader: constants_1.default.loadingMode.BLUR,
                placementAnimation: constants_1.default.imagePlacementAnimations.NO_EFFECT,
                magnificationValue: 2,
            },
        },
        gallery: {
            blockContextMenu: true,
            layoutDirection: 'LEFT_TO_RIGHT',
            scrollAnimation: constants_1.default.scrollAnimations.NO_EFFECT,
            enableIndexingShareLinks: true,
            horizontal: {
                enableScrollSnap: false,
                navigationDuration: 400,
                blockScroll: false,
                loop: false,
                slideTransition: constants_1.default.slideTransition.ease,
                slideAnimation: constants_1.default.slideAnimations.SCROLL,
                autoSlide: {
                    behaviour: 'OFF',
                    interval: 4,
                    pauseOnHover: true,
                    speed: 200,
                },
                slideshowInfo: {
                    buttonsAlignment: 'CENTER',
                    enablePlayButton: false,
                    enableCounter: false,
                },
            },
            vertical: {
                loadMore: {
                    enable: false,
                    text: '',
                    amount: 'ALL',
                },
            },
        },
    },
    stylingParams: {
        itemBorderWidth: 0,
        itemBorderRadius: 0,
        itemEnableShadow: false,
        itemShadowBlur: 20,
        itemShadowDirection: 135,
        itemShadowSize: 10,
        itemShadowOpacityAndColor: '',
        arrowsColor: '',
    },
};
exports.default = defaultOptions;
