// pro-gallery-lib/dist/cjs/settings/options/arrowsVerticalPosition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = require("../utils/constants");
var constants_2 = tslib_1.__importDefault(require("../../common/constants"));
var utils_1 = require("../utils/utils");
var arrowsPosition_1 = tslib_1.__importDefault(require("./arrowsPosition"));
exports.default = {
    title: 'Navigation Arrows Vertical Position',
    isRelevant: function (options) {
        return arrowsPosition_1.default.isRelevant(options) &&
            (options.arrowsPosition === constants_2.default.arrowsPosition.ON_GALLERY ||
                options.arrowsPosition === constants_2.default.arrowsPosition.OUTSIDE_GALLERY);
    },
    isRelevantDescription: arrowsPosition_1.default.isRelevantDescription +
        ' and make sure arrow position is set to "on gallery" or "outside gallery"',
    type: constants_1.INPUT_TYPES.OPTIONS,
    default: constants_2.default.arrowsVerticalPosition.ITEM_CENTER,
    options: utils_1.createOptions('arrowsVerticalPosition'),
    description: "Set the vertical position of the navigation arrows in sliders. You can choose to position the arrows on the center of the whole item, the center of the image or the center of the info",
};
