// pro-gallery-lib/dist/cjs/settings/options/behaviourParams_item_secondaryMedia_behaviour.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = require("../utils/constants");
var constants_2 = tslib_1.__importDefault(require("../../common/constants"));
var utils_1 = require("../utils/utils");
var behaviourParams_item_secondaryMedia_trigger_1 = tslib_1.__importDefault(require("./behaviourParams_item_secondaryMedia_trigger"));
exports.default = {
    title: 'Second Media Behaivour',
    isRelevant: function (sp) {
        return behaviourParams_item_secondaryMedia_trigger_1.default.isRelevant(sp) &&
            sp.behaviourParams.item.secondaryMedia.trigger !==
                constants_2.default.secondaryMediaTrigger.OFF;
    },
    isRelevantDescription: 'Secondary Media Trigger is not "OFF" and when arrowsPosition is MOUSE_CURSOR there must be some space to hover (mouseCursorContainerMaxWidth less then 100)',
    type: constants_1.INPUT_TYPES.OPTIONS,
    default: constants_2.default.secondaryMediaBehaviour.APPEARS,
    options: utils_1.createOptions('secondaryMediaBehaviour'),
    description: "Select the behaviour for the second media on action.",
};
