// pro-gallery-lib/dist/cjs/settings/options/behaviourParams_item_secondaryMedia_trigger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = require("../utils/constants");
var constants_2 = tslib_1.__importDefault(require("../../common/constants"));
var utils_1 = require("../utils/utils");
var arrowsPosition_1 = tslib_1.__importDefault(require("./arrowsPosition"));
var layoutParams_navigationArrows_mouseCursorContainerMaxWidth_1 = tslib_1.__importDefault(require("./layoutParams_navigationArrows_mouseCursorContainerMaxWidth"));
exports.default = {
    title: 'Second Media Trigger',
    isRelevant: function (options) {
        return !(arrowsPosition_1.default.isRelevant(options) &&
            options.arrowsPosition === constants_2.default.arrowsPosition.MOUSE_CURSOR &&
            options.layoutParams.navigationArrows.mouseCursorContainerMaxWidth ===
                100 &&
            layoutParams_navigationArrows_mouseCursorContainerMaxWidth_1.default.isRelevant(options));
    },
    isRelevantDescription: 'Not relevant when arowsPosition is MOUSE_CURSOR and mouseCursorContainerMaxWidth is 100',
    type: constants_1.INPUT_TYPES.OPTIONS,
    default: constants_2.default.secondaryMediaTrigger.OFF,
    options: utils_1.createOptions('secondaryMediaTrigger'),
    description: "Select the triggering action that will show the second media.",
};
