// pro-gallery-lib/dist/cjs/settings/options/itemClick.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = require("../utils/constants");
var utils_1 = require("../utils/utils");
var constants_2 = tslib_1.__importDefault(require("../../common/constants"));
var layoutParams_navigationArrows_mouseCursorContainerMaxWidth_1 = tslib_1.__importDefault(require("./layoutParams_navigationArrows_mouseCursorContainerMaxWidth"));
exports.default = {
    title: 'Click Action',
    isRelevant: function (options) {
        return !(options.layoutParams.navigationArrows.mouseCursorContainerMaxWidth ===
            100 &&
            layoutParams_navigationArrows_mouseCursorContainerMaxWidth_1.default.isRelevant(options));
    },
    isRelevantDescription: 'its always relevant except if arrow position is set to mouse cursor and the mouse cursor container max width is 100 (clicks will navigate to the next item)',
    type: constants_1.INPUT_TYPES.OPTIONS,
    default: constants_2.default.itemClick.NOTHING,
    options: utils_1.createOptions('itemClick'),
    description: "Specifies what happens when an item is clicked. To enable the 'expand' or 'fullscreen' options, make sure you're using the ExpandableProGallery component",
};
