// pro-gallery-lib/dist/cjs/settings/options/layoutParams_info_height.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = require("../utils/constants");
var constants_2 = tslib_1.__importDefault(require("../../common/constants"));
exports.default = {
    title: 'Text Box Height (Pixels)',
    description: "Set the text box Height when the texts are above or below the item",
    isRelevantDescription: 'First set a Vertical gallery ("Scroll Direction" as "Vertical").\nThen set "Layout Orientation" to "Columns".\nThen set "Max Group Size" to "1".\nThen set "Texts Placement" to "Show Below" or "Show Above".',
    isRelevant: function (options) {
        return ((options.isVertical &&
            options.scrollDirection === constants_2.default.scrollDirection.VERTICAL) ||
            options.scrollDirection === constants_2.default.scrollDirection.HORIZONTAL) &&
            options.groupSize === 1 &&
            constants_2.default.hasExternalVerticalPlacement(options.titlePlacement);
    },
    type: constants_1.INPUT_TYPES.NUMBER,
    default: 200,
    min: 0,
    max: 1000,
};
