// pro-gallery-lib/dist/cjs/settings/options/layoutParams_info_width.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var constants_1 = require("../utils/constants");
var constants_2 = tslib_1.__importDefault(require("../../common/constants"));
exports.default = {
    title: 'Text Box Width',
    description: "Set the text box width (accourding to the selected unit) from the column width when on the right side or on the left side.",
    isRelevant: function (options) {
        return options.isVertical &&
            options.groupSize === 1 &&
            options.scrollDirection === constants_2.default.scrollDirection.VERTICAL &&
            constants_2.default.hasExternalHorizontalPlacement(options.titlePlacement);
    },
    isRelevantDescription: "First set a Vertical gallery (\"Scroll Direction\" as \"Vertical\").\nThen set \"Layout Orientation\" to \"Columns\".\nThen set \"Max Group Size\" to \"1\".\nThen set \"Texts Placement\" to \"Show on the Right\" or \"Show on the Left\".",
    type: constants_1.INPUT_TYPES.NUMBER,
    default: 50,
    min: 0,
    max: 100,
};
