// pro-gallery/dist/esm/components/gallery/typeValidator/typeErrorsUI.js
import { __extends } from "tslib";
import React from 'react';
export default function (typeErrors) {
    window.typeErrors = typeErrors;
    return (React.createElement("div", null,
        React.createElement("ol", null, typeErrors.map(function (errObject, i) { return (React.createElement(Error, { errObject: errObject, key: i })); })),
        JSON.stringify(typeErrors, null, 4)));
}
function Error(_a) {
    var errObject = _a.errObject;
    return (React.createElement("li", null,
        React.createElement(ErrorDescription, { errObject: errObject })));
}
var ErrorDescription = /** @class */ (function (_super) {
    __extends(ErrorDescription, _super);
    function ErrorDescription(props) {
        var _this = _super.call(this, props) || this;
        _this.state = {
            showFull: false,
        };
        _this.toggle = _this.toggle.bind(_this); // PUKE
        return _this;
    }
    ErrorDescription.prototype.render = function () {
        var CompToRender = this.state.showFull ? FullError : Text;
        return (React.createElement("a", { onClick: this.toggle },
            React.createElement(CompToRender, { errObject: this.props.errObject })));
    };
    ErrorDescription.prototype.toggle = function () {
        this.setState({ showFull: !this.state.showFull });
    };
    return ErrorDescription;
}(React.Component));
function FullError(_a) {
    var errObject = _a.errObject;
    return (React.createElement("ul", null, Object.entries(errObject).map(function (entry, i) { return (React.createElement(Item, { entry: entry, key: i })); })));
}
function Item(_a) {
    var entry = _a.entry;
    var key = entry[0];
    var value = entry[1];
    var valueText = typeof value === 'string' ? value : JSON.stringify(value, null, 4);
    return React.createElement("li", null, key + ": " + valueText);
}
function Text(_a) {
    var errObject = _a.errObject;
    return errObject.message;
}
